src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -363,6 +452,13 @@ protected override void VisitSwitchSection(BoundSwitchSection node, bool isLastS
                                         }
                                         Debug.Assert(outputSlot > 0);
                                         addToTempMap(output, outputSlot, type.Type);
+
+                                        if (property.GetMethod is not null)
+                                        {
+                                            // A property evaluation splits the state if MemberNotNullWhen is used
+                                            ApplyMemberPostConditions(inputSlot, property.GetMethod);
```

Comment:
I'm confused about this bit in light of the `gotoNodeWithCurrentStae` after it. It seems like when you apply the member postconditions, you can get into a split state, but then we will `gotoNodeWithCurrentState`, which expects a non-split state. What am I missing?

Reply:
`gotoNodeWithCurrentState` can handle split states, but it currently asserts out if we have a split state *and* we're merging two branches (the next node already has some accumulated state attached to it). I've tried a bunch of things but could not hit that combination.

---
In reply to: [574024479](https://github.com/dotnet/roslyn/pull/51001#discussion_r574024479) [](ancestors = 574024479)